client/core/core_test.go

```diff
@@ -352,7 +353,32 @@ func (tdb *TDB) LinkOrder(oid, linkedID order.OrderID) error {
 	return nil
 }
 
+func (tdb *TDB) setUpdateMatchHook(ch chan order.MatchStatus) {
+	tdb.updateMatchMtx.Lock()
+	defer tdb.updateMatchMtx.Unlock()
+	tdb.updateMatchChan = ch
+}
+
+func (tdb *TDB) waitForMatchUpdate() (order.MatchStatus, error) {
+	tdb.updateMatchMtx.RLock()
+	defer tdb.updateMatchMtx.RUnlock()
+	var latestStatus order.MatchStatus
+	if tdb.updateMatchChan == nil {
+		return latestStatus, errors.New("not configured to track match db updates")
+	}
+	select {
+	case latestStatus = <-tdb.updateMatchChan:
+		return latestStatus, nil
+	case <-time.After(1 * time.Second):
+		return latestStatus, errors.New("no match db update in 1 second")
+	}
```

Comment:
I also ran into a situation where the test previously expected one db update for setting both `{M,T}aker{SwapCast,Redeemed}` statuses AND `auth.{Init,Redeem}Sig`. Because only one db update was required, the `tdb.updateMatchChan` was only drained once but the changes in this PR now triggers 2 db updates causing the channel to block when a third db update call was triggered by some other operation.

The test timed out before I could determine that the fix was to call `tdb.waitForMatchUpdate()` twice rather than once for init and redeem scenarios. This change makes it easier and quicker to catch this kind of issue subsequently.